testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPVaultCredentialsTest.java

```diff
@@ -0,0 +1,33 @@
+package org.keycloak.testsuite.federation.ldap;
+
+import org.junit.ClassRule;
+import org.keycloak.testsuite.util.LDAPRule;
+
+import java.util.Map;
+
+import static org.keycloak.models.LDAPConstants.BIND_CREDENTIAL;
+
+/**
+ * @author mhajas
+ */
+public class LDAPVaultCredentialsTest extends LDAPSyncTest {
+
+    private static final String VAULT_EXPRESSION = "${vault.ldap_bindCredential}";
+
+    @ClassRule
+    public static LDAPRule ldapRule = new LDAPRule() {
```

Comment:
LDAP tests can be executed for some other LDAP servers from the lab and QA environments and in those tests, the bind_credential to the LDAP server can be something different than "secret" . I guess it won't work with the current vault right now?

Since this PR doesn't add any LDAP protocol specific stuff, but it rather adds "keycloak integration" stuff, it may be ok to test it just with the embedded ApacheDS LDAP server? See for example definition of LDAPRule in LDAPNoMSADTest how to do something like this.

Reply:
Fixed. I added a check for "secret" password in the configuration. Do you think this is correct approach?

Reply:
That can work. Maybe even slightly better could be to check if embedded LDAP server is used? Something like:
```
.assumeTrue((LDAPTestConfiguration ldapConfig) -> {
        return ldapConfig.isEmbeddedLdapServer();        
    });
```
Otherwise, if we somehow change the password of embedded LDAP server from "secret" to something else, we will just stop testing LDAP with the vault and nobody will notice :) Not sure if we ever change the password, but rather slightly prefer this.

Reply:
Fixed

Reply:
Thanks!